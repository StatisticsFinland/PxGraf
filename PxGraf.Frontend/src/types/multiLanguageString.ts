/* istanbul ignore file */

/**
 * Type for string supporting multiple languages with each key representing a language and the value the string in that language.
 */
export type MultiLanguageString = { [key: string]: string };