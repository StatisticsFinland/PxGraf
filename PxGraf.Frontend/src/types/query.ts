/* istanbul ignore file */

/* Query - dimension queries*/

/**
 * Type for a query that contains @see {@link IDimensionQuery} objects for each dimension.
 * @property {{ [key: string]: IDimensionQuery }} - Queries for each dimension.
 */
export type Query = { [key: string]: IDimensionQuery };

/**
 * Enum that defines different dimension value filter types.
 */
export enum FilterType {
    Item = 'item',
    All = 'all',
    Top = 'top',
    From = 'from'
}

/**
 * Interface for dimension query properties.
 * @property {IValueFilter} valueFilter - Reference to value filter.
 * @property {boolean} selectable - Selected selectable dimension.
 * @property {any} virtualValueDefinitions - Virtual value definitions.
  */
export interface IDimensionQuery {
    valueFilter: IValueFilter
    selectable: boolean
    virtualValueDefinitions
}

/**
 * Interface for value filter properties.
 * @property {FilterType} type - Selected filter type.
 * @property {string[] | string | number} query - The query for the filter.
 */
export interface IValueFilter {
    type: FilterType;
    query?: string[] | string | number
}

/**
 * Type for string supporting multiple languages.
 */
type MultiLanguageString = { [key: string]: string };

/* CubeQuery - meta data editions*/

/**
 * Interface for cube query properties.
 * @property {MultiLanguageString} chartHeaderEdit - The chart header edition.
 * @property {{ [key: string]: IDimensionEditions }} variableQueries - The IDimensionEditions edition queries.
 */
export interface ICubeQuery {
    chartHeaderEdit?: MultiLanguageString;
    variableQueries: { [key: string]: IDimensionEditions };
}

/**
 * Interface for IDimensionEditions editions.
 * @property {{ [key: string]: IDimensionValueEditions }} valueEdits - The value editions.
 */
export interface IDimensionEditions {
    valueEdits: {
        [valueCode: string]: IDimensionValueEditions;
    };
}

/**
 * Interface for dimension value editions.
 * @property {MultiLanguageString} nameEdit - The name edition.
 * @property {IContentComponentEdition} contentComponent - The content component edition containing source and unit.
  */
export interface IDimensionValueEditions {
    nameEdit? : MultiLanguageString;
    contentComponent? : IContentComponentEdition;
}

/**
 * Interface for content component edition.
 * @property {MultiLanguageString} sourceEdit - Edition of the source.
 * @property {MultiLanguageString} unitEdit - Edition of the unit.
 */
interface IContentComponentEdition {
    sourceEdit? : MultiLanguageString;
    unitEdit? : MultiLanguageString;
}

/**
 * Interface that stores information about the query.
 * @property {number} size - The size of the query.
 * @property {number} sizeWarningLimit - The size warning limit of the query.
 * @property {number} maximumSupportedSize - The maximum supported size of the query.
 * @property {string[]} validVisualizations - Names of valid visualization types for of the query.
 * @property {{[key: string]: MultiLanguageString}} visualizationRejectionReasons - Reasons for why certain visualization types are rejected for the query.
 * @property {number} maximumHeaderLength - The maximum header length of the query.
 */
export interface IQueryInfo {
    size: number;
    sizeWarningLimit: number;
    maximumSupportedSize: number;
    validVisualizations: string[];
    visualizationRejectionReasons: {[key: string]: MultiLanguageString};
    maximumHeaderLength: number;
}