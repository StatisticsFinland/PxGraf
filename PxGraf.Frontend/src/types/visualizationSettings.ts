/* istanbul ignore file */

import {VisualizationType} from './visualizationType';

/**
 * Interface for visualization settings
 * @property {string[]} defaultSelectableVariableCodes - Dimension codes for default selectable dimension values.
 * @property {boolean} pivotRequested - Flag to indicate if the user has requested manual pivoting.
 * @property {boolean} cutYAxis - Flag to indicate if the y-axis should be cut based on the values.
 * @property {string} multiselectableVariableCode - The dimension code for the multiselectable dimension if any.
 * @property {string[]} rowVariableCodes - Dimension codes for the row dimensions.
 * @property {string[]} columnVariableCodes - Dimension codes for the column dimensions.
 * @property {string} sorting - Selected sorting order.
 * @property {boolean} matchXLabelsToEnd - Flag to indicate if the x-axis labels should be matched to the end.
 * @property {number} markerSize - The size of scatter plot markers.
 * @property {VisualizationType} selectedVisualization - The selected visualization type.
 * @property {boolean} showDataPoints - Flag to indicate if data points should be shown in the visualization.
 */
export interface IVisualizationSettings {
    defaultSelectableVariableCodes?: { [key: string]: string[] };
    pivotRequested?: boolean;
    cutYAxis?: boolean;
    multiselectableVariableCode?: string;
    rowVariableCodes?: string[];
    columnVariableCodes?: string[];
    sorting?: string;
    matchXLabelsToEnd?: boolean;
    markerSize?: number;
    selectedVisualization?: VisualizationType;
    showDataPoints?: boolean;
}