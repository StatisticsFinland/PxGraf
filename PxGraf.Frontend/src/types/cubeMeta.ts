/* istanbul ignore file */

import { MultiLanguageString } from "./multiLanguageString";

/**
 * Interface for cube meta properties. A cube is a multi-dimensional result of Px file data filtered by a query.
 * @property {string[]} languages - List of available languages.
 * @property {MultiLanguageString} header - The multi language header of the cube.
 * @property {MultiLanguageString} note - The note of the cube.
 * @property {IVariable[]} variables - List of variables.
 */
export interface IMatrixMetadata {
    DefaultLanguage: string,
    AvailableLanguages: string[],
    Dimensions: IDimension[],
    AdditionalProperties?: { [key: string]: IMetaProperty }
}

/**
 * TODO: Summary
 */
export interface IMetaProperty {
    KeyWord: string,
    CanGetStringValue: boolean,
    CanGetMultilanguageValue: boolean,
    Entries: MultiLanguageString | string
}

/**
 * Interface for dimnension properties. Each variable defines a dimension in a cube.
 * @property {string} code - The code of the variable.
 * @property {VariableType} type - The type of the variable.
 * @property {MultiLanguageString} name - The multi language name of the variable.
 * @property {{ [key: string]: IMetaProperty }} additionalProperties - Optional dictionary for additional metadata properties
 * @property {IVariableValue[]} values - List of values associated with this variable.
 */
export interface IDimension {
    Code: string,
    Name: MultiLanguageString,
    AdditionalProperties?: { [key: string]: IMetaProperty },
    Values: IDimensionValue[]
    Type: VariableType,
}

/**
 * Interface for variable properties.Each variable defines a dimension in a cube.
 * @property { string } code - The code of the variable.
 * @property { MultiLanguageString } name - The multi language name of the variable.
 * @property { MultiLanguageString } note - The multi language note of the variable.
 * @property { VariableType } type - The type of the variable.
 * @property { IVariableValue[] } values - List of values associated with this variable.
 */
export interface IVariable {
    code: string,
    name: MultiLanguageString,
    note: MultiLanguageString,
    type: VariableType,
    values: IVariableValue[]
}

/**
 * Interface for dimension value properties.
 * @property {string} code - The code of the value.
 * @property {MultiLanguageString} name - The multi language name of the value.
 * @property {{ [key: string]: IMetaProperty }} additionalProperties - Optional dictionary for additional metadata properties
 */
export interface IDimensionValue {
    Code: string,
    Name: MultiLanguageString,
    Virtual: boolean
    AdditionalProperties?: { [key: string]: IMetaProperty },
}

/**
 * Interface for variable value properties.
 * @property { string } code - The code of the value.
 * @property { MultiLanguageString } name - The multi language name of the value.
 * @property { MultiLanguageString } note - The multi language note of the value.
 * @property { boolean } isSum - Flag to indicate if the value is an elimination value.
 * @property { IContentComponent } contentComponent - The content variable component of the value.
 * @property { IVirtualComponent[] } virtualComponents - List of virtual components.
 */
export interface IVariableValue {
    code: string,
    name: MultiLanguageString,
    note: MultiLanguageString,
    isSum: boolean,
    contentComponent?: IContentComponent
    virtualComponents?: IVirtualComponent[]
}

/**
 * Interface for value content variable value properties.
 * @property {MultiLanguageString} unit - The unit of the value.
 * @property {MultiLanguageString} source - The source of the value.
 * @property {number} numberOfDecimals - The precision, number of decimals of the value.
 * @property {string} lastUpdated - The last updated date of the value.
 */
export interface IContentComponent {
    unit: MultiLanguageString,
    source: MultiLanguageString,
    numberOfDecimals: number,
    lastUpdated: string
}

/**
 * Interface for virtual component properties.
 * @property {string} operator - The operator of the virtual component.
 * @property {string[]} operandCoded - List of coded operands.
 */
export interface IVirtualComponent {
    operator: string,
    operandCoded: string[]
}

/**
 * Enumeration for variable types.
 */
export enum VariableType {
    Unknown = 'N',
    Time = 'T',
    Ordinal = 'P',
    Geographical = 'G',
    Content = 'C',
    OtherClassificatory = 'F',
}

export function getVariablesFromMatrixMetadata(meta: IMatrixMetadata): IVariable[] {
    return meta.Dimensions.map(d => {
        return {
            code: d.Code,
            name: d.Name,
            note: d.Name,
            type: d.Type,
            values: d.Values.map(v => convertDimensionValueToVariableValue(v, meta))
        }
    })
}

export function convertDimensionValueToVariableValue(value: IDimensionValue, meta: IMatrixMetadata): IVariableValue {
    const dimension: IDimension = meta.Dimensions.find(d => d.Values.find(v => v.Code === value.Code));
    const contentComponent = (): IContentComponent | null => {
        if (dimension.Type === VariableType.Content) {
            return {
                unit: getAdditionalProperty("UNIT", value.AdditionalProperties) as MultiLanguageString,
                source: getAdditionalProperty("SOURCE", value.AdditionalProperties) as MultiLanguageString,
                numberOfDecimals: parseInt(getAdditionalProperty("PRECISION", value.AdditionalProperties, true) as string),
                lastUpdated: getAdditionalProperty("LAST-UPDATED", value.AdditionalProperties, true) as string
            }
        }
        else return null;
    }
    return {
        code: value.Code,
        name: value.Name,
        note: value.Name,
        isSum: getValueIsSumValue(value, dimension),
        contentComponent: contentComponent()
    }
}

export function getValueIsSumValue(value: IDimensionValue, dimension: IDimension): boolean {
    return getAdditionalProperty("ELIMINATION", dimension.AdditionalProperties) === value.Code;
}

function getAdditionalProperty(key: string, additionalProperties: { [key: string]: IMetaProperty }, forceToString = false): string | MultiLanguageString {
    if (additionalProperties[key]) {
        if (additionalProperties[key].CanGetStringValue) {
            return additionalProperties[key].Entries[0];
        }
        else if (additionalProperties[key].CanGetMultilanguageValue) {
            if (forceToString) return additionalProperties[key].Entries[0];
            else return additionalProperties[key].Entries;
        }
    }
    return null;
}
