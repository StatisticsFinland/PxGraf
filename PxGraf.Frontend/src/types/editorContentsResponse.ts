import { MultiLanguageString } from "./multiLanguageString";
import { VisualizationType } from "./visualizationType";

/**
 * Interface for dimension sorting options.
 * @property {string} code - The code of the sorting option.
 * @property {MultiLanguageString} description - Longer, user facing description of the sorting option.
 */
export interface ISortingOption {
    code: string;
    description: MultiLanguageString;
}

/**
 * Interface for sorting option collection. Contains default and pivoted sorting options if available.
 * @property {ISortingOption[]} default - Default sorting options.
 * @property {ISortingOption[]} pivoted - Pivoted sorting options.
 */
interface ISortingOptionCollection {
    default: ISortingOption[];
    pivoted: ISortingOption[];
}

/**
 * Interface for visualization options for a specific visualization type.
 * @property {VisualizationType} type - The type of visualization.
 * @property {boolean} [allowShowingDataPoints] - Flag to indicate if showing data points is allowed.
 * @property {boolean} [allowCuttingYAxis] - Flag to indicate if cutting the Y-axis is allowed.
 * @property {boolean} [allowMatchXLabelsToEnd] - Flag to indicate if matching X labels to the end is allowed.
 * @property {boolean} [allowSetMarkerScale] - Flag to indicate if setting marker scale is allowed.
 * @property {boolean} allowManualPivot - Flag to indicate if manual pivoting is allowed.
 * @property {boolean} allowMultiselect - Flag to indicate if multivalue selectable is allowed.
 * @property {ISortingOptionCollection} sortingOptions - Sorting options for the visualization containing default and pivoted options.
 */
export interface IVisualizationOptions {
    type: VisualizationType;
    allowShowingDataPoints?: boolean;
    allowCuttingYAxis?: boolean;
    allowMatchXLabelsToEnd?: boolean;
    allowSetMarkerScale?: boolean;
    allowManualPivot: boolean;
    allowMultiselect: boolean;
    sortingOptions: ISortingOptionCollection;
}

/**
 * Interface for the response from the editor contents API used to populate the settings for adjusting the visualization in the editor.
 * @property {number} size - The size of the response based on the amount of values selected from all dimensions.
 * @property {number} sizeWarningLimit - The size warning limit for the response.
 * @property {number} maximumSupportedSize - The maximum supported size for the response.
 * @property {MultiLanguageString} headerText - Default header text for the visualization as a @ {@link MultiLanguageString} object.
 * @property {number} maximumHeaderLength - The maximum length of the header text.
 * @property {IVisualizationOptions[]} visualizationOptions - Visualization options for the valid visualization types as an array of @ {@link IVisualizationOptions} objects.
 * @property {object} visualizationRejectionReasons - Object containing rejection reasons for each visualization type as a @ {@link MultiLanguageString} object.
 */
export interface IEditorContentsResponse {
    size: number;
    sizeWarningLimit: number;
    maximumSupportedSize: number;
    headerText: MultiLanguageString;
    maximumHeaderLength: number;
    visualizationOptions: IVisualizationOptions[];
    visualizationRejectionReasons: {
        [visualizationType: string]: MultiLanguageString;
    }
}