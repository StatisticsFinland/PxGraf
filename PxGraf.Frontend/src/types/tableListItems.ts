/* istanbul ignore file */

import { MultiLanguageString } from "./multiLanguageString";

/**
 * Interface for a section in the hierarchy of the table list.
 * @property {IDatabaseGroupHeader[]} headers - The headers in the section.
 * @property {IDatabaseTable[]} files - The tables in the section.
 */
export interface IDatabaseGroupContents {
    headers: IDatabaseGroupHeader[];
    files: IDatabaseTable[];
}

/**
 * Interface for items shown on the table listing that can be sorted by the translated name.
 * @property {MultiLanguageString} name - Localized names in all available languages.
 * @property {string[]} languages - Available languages of the item. 
 */
export interface ISortableTableListItem {
    name: MultiLanguageString;
    languages: string[];
}

/***
 * Interface for a table group header
 * @property {string} code - The code of the table group.
 * @property {MultiLanguageString} name - The name of the table group as a multi-language string.
 */
export interface IDatabaseGroupHeader extends ISortableTableListItem {
    code: string;
}

/***
 * Enum for database table errors
 */
export enum EDatabaseTableError {
    contentLoad = 'ContentLoad',
    contentDimensionMissing = 'ContentDimensionMissing',
    timeDimensionMissing = 'TimeDimensionMissing',
}

/***
 * Interface for a px table
 * @property {fileName} fileName - Name of the px table file.
 * @property {string} lastUpdated - The last updated date of the table.
 * @property {EDatabaseTableError} error - Enum to indicate if there's an error with the table.
 */
export interface IDatabaseTable extends ISortableTableListItem {
    fileName: string;
    lastUpdated: string | null;
    error?: EDatabaseTableError;
}
