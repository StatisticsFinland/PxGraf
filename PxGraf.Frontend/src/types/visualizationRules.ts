/* istanbul ignore file */

import { MultiLanguageString } from "./multiLanguageString"

/**
 * Interface for visualization rules.
 * @property allowManualPivot - Flag to indicate if manual pivot is allowed.
 * @property sortingOptions - List of sorting options.
 * @property multiselectVariableAllowed - Flag to indicate if multiselect variable is allowed.
 * @property visualizationTypeSpecificRules - Object for visualization type specific rules.
 */
export interface IVisualizationRules {
    allowManualPivot: boolean,
    sortingOptions: ISortingOption[],
    multiselectVariableAllowed: boolean
    visualizationTypeSpecificRules?: ITypeSpecificVisualizationRules
}

/**
 * Interface for different sorting options
 * @property code - Code for the sorting option
 * @property description - Description for the sorting option as a multi language string
 */
export interface ISortingOption {
    code: string,
    description: MultiLanguageString
}

/**
 * Interface for visualization type specific rules.
 * @property allowShowingDataPoints - Flag to indicate if showing data points is allowed.
 * @property allowCuttingYAxis - Flag to indicate if cutting the Y-axis is allowed.
 * @property allowMatchXLabelsToEnd - Flag to indicate if matching X labels to the end is allowed.
 * @property allowSetMarkerScale - Flag to indicate if setting marker scale is allowed.
 */
export interface ITypeSpecificVisualizationRules {
    allowShowingDataPoints: boolean,
    allowCuttingYAxis: boolean,
    allowMatchXLabelsToEnd: boolean,
    allowSetMarkerScale: boolean
}