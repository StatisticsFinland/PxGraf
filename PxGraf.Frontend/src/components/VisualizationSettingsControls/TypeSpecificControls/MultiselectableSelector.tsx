import React from 'react';
import { FormControl, Select, InputLabel, MenuItem} from '@mui/material';
import { IVisualizationSettingsProps } from '../VisualizationSettingsControl';
import { IVariable } from "types/cubeMeta";
import { UiLanguageContext } from 'contexts/uiLanguageContext';
import { useTranslation } from 'react-i18next';

interface MultiselectableSelectorProps extends IVisualizationSettingsProps {
    variables: IVariable[]
}

export const MultiselectableSelector: React.FC<MultiselectableSelectorProps> = ({ settingsChangedHandler, visualizationSettings, variables  }) => {
    const { t } = useTranslation();
    const { language, languageTab } = React.useContext(UiLanguageContext);

    return (
        <FormControl>
            <InputLabel id="multiselectable-selector-label">{t("chartSettings.multiSelectVariable")}</InputLabel>
            <Select
                sx={{ minWidth: 210 }}
                labelId="multiselectable-selector-label"
                id="multiselectable-selector"
                label={"Monivalitaselausmuuttuja"}
                value={visualizationSettings.multiselectableVariableCode ?? "noMultiselectable"}
                defaultValue={"noMultiselectable"}
                onChange={(event) => settingsChangedHandler({
                    ...visualizationSettings,
                    multiselectableVariableCode: event.target.value !== "noMultiselectable" ? event.target.value : null
                })}
            >
                <MenuItem value={"noMultiselectable"}>{t("chartSettings.noMultiselectable")}</MenuItem>
                {variables.map(v => { return <MenuItem key={"key" + v.code} value={v.code}>{v.name[language] ?? v.name[languageTab] ?? v.code}</MenuItem> })}
            </Select>
        </FormControl>
    )
}