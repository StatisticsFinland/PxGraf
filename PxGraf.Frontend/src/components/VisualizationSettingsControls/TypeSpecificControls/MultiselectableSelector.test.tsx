import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { MultiselectableSelector } from './MultiselectableSelector';
import { IDimension, EDimensionType } from "types/cubeMeta";
import { IVisualizationSettings } from '../../../types/visualizationSettings';
import { IVisualizationRules } from '../../../types/visualizationRules';

const mockDimensions: IDimension[] = [
	{
		code: 'cVar',
		name: { fi: 'cVarName' },
		type: EDimensionType.Content,
		values: [
			{
				code: 'cVal',
				name: { fi: 'cValName' },
				isVirtual: false
			}
		]
	},
	{
		code: 'tVar',
		name: { fi: 'tVarName' },
		type: EDimensionType.Time,
		values: [
			{
				code: 'tVal',
				name: { fi: 'tValName' },
				isVirtual: false
			}
		]
	},
	{
		code: 'msVar',
		name: { fi: 'msVarAName' },
		type: EDimensionType.Other,
		values: [
			{
				code: 'msVal1',
				name: { fi: 'msVal1Name' },
				isVirtual: false
			},
			{
				code: 'msVal2',
				name: { fi: 'msVal2Name' },
				isVirtual: false
			}
		]
	}
];

const mockVisualizationSettings: IVisualizationSettings = {
	multiselectableVariableCode: 'msVar'
};
const mockVisualizationRules: IVisualizationRules = {
	allowManualPivot: false, sortingOptions: null, multiselectDimensionAllowed: true
};

jest.mock('react-i18next', () => ({
	...jest.requireActual('react-i18next'),
	useTranslation: () => {
		return {
			t: (str: string) => str,
			i18n: {
				changeLanguage: () => new Promise(() => null),
			},
		};
	},
}));

describe('Rendering test', () => {
	it('renders correctly', () => {
		const { asFragment } = render(
			<MultiselectableSelector
				visualizationRules={mockVisualizationRules}
				visualizationSettings={mockVisualizationSettings}
				dimensions={mockDimensions}
			/>);
		expect(asFragment()).toMatchSnapshot();
	});
});



describe('Assertion tests', () => {
	it('When no multiselect variable code is provided, the selector should default to "noMultiselectable"', () => {
		render(<MultiselectableSelector
			dimensions={mockDimensions}
			visualizationRules={mockVisualizationRules}
			visualizationSettings={{ ...mockVisualizationSettings, multiselectableVariableCode: null }}
		></MultiselectableSelector>);
		expect(screen.getByDisplayValue("noMultiselectable")).toBeInTheDocument();
	});

	it('When multiselect variable code is provided, the selector should be rendered with a corresponding value', () => {
		const changedMockSettings = { ...mockVisualizationSettings, multiselectableDimensionCode: "msVar" }
		render(<MultiselectableSelector
			dimensions={mockDimensions}
			visualizationRules={mockVisualizationRules}
			visualizationSettings={changedMockSettings}
		></MultiselectableSelector>);
		expect(screen.queryByDisplayValue("msVar")).toBeInTheDocument();
	});
});