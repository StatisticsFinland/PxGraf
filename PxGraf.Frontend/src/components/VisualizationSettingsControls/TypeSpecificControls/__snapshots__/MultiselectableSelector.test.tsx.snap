// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering test renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      id="multiselectable-selector-label"
    >
      chartSettings.multiSelectVariable
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1l9f323-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    >
      <div
        aria-controls=":r0:"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="multiselectable-selector-label multiselectable-selector"
        class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        id="multiselectable-selector"
        role="combobox"
        tabindex="0"
      >
        msVar
      </div>
      <input
        aria-hidden="true"
        aria-invalid="false"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        tabindex="-1"
        value="msVar"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-14lo706"
        >
          <span>
            Monivalitaselausmuuttuja
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
